test/pytest/mocktransport.py

```diff
@@ -13,7 +13,8 @@ class MockTransport(asyncio.Transport):
     close_future: Optional[asyncio.Future] = None
 
     def send_json(self, _channel: str, **kwargs) -> None:
-        msg = {k.replace('_', '-'): v for k, v in kwargs.items()}
+        # max_read_size is one of our special keys which uses underscores
+        msg = {k.replace('_', '-') if k != "max_read_size" else k: v for k, v in kwargs.items()}
```

Comment:
I'd prefer to avoid the hack here.  Can you just use `.send_data()` from the call site?

Reply:
It is a bit more elaborate but I can make it happen :)